test/OpenFeature.Tests/OpenFeatureEventTests.cs

```diff
@@ -0,0 +1,255 @@
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+using System.Threading.Tasks;
+using AutoFixture;
+using NSubstitute;
+using OpenFeature.Constant;
+using OpenFeature.Model;
+using Xunit;
+
+namespace OpenFeature.Tests
+{
+    [SuppressMessage("Reliability", "CA2007:Consider calling ConfigureAwait on the awaited task")]
+    public class OpenFeatureEventTests : ClearOpenFeatureInstanceFixture
+    {
+        [Fact]
+        public async Task Event_Executor_Should_Propagate_Events_ToGlobal_Handler()
+        {
+            var eventHandler = Substitute.For<EventHandlerDelegate>();
+
+            eventHandler.Invoke(Arg.Any<ProviderEventPayload>());
+
+            var eventExecutor = new EventExecutor();
+
+            eventExecutor.AddApiLevelHandler(ProviderEventTypes.ProviderReady, eventHandler);
+
+            var eventPayload = new Event { EventPayload = new ProviderEventPayload { Type = ProviderEventTypes.ProviderReady } };
+            eventExecutor.EventChannel.Writer.TryWrite(eventPayload);
+
+            Thread.Sleep(1000);
+
+            eventHandler.Received().Invoke(Arg.Is<ProviderEventPayload>(payload => payload.Type == ProviderEventTypes.ProviderReady));
+
+            // shut down the event executor
+            await eventExecutor.SignalShutdownAsync();
+
+            // the next event should not be propagated to the event handler
+            var newEventPayload = new ProviderEventPayload { Type = ProviderEventTypes.ProviderStale };
+
+            eventExecutor.EventChannel.Writer.TryWrite(newEventPayload);
+
+            eventHandler.DidNotReceive().Invoke(newEventPayload);
+
+            eventHandler.DidNotReceive().Invoke(Arg.Is<ProviderEventPayload>(payload => payload.Type == ProviderEventTypes.ProviderStale));
+        }
+
+        [Fact]
```

Comment:
Are there any of these that can be tied back to their specification?

Example from other tests:
```
[Specification("1.1.2.2", "The provider mutator function MUST invoke the initialize function on the newly registered provider before using it to resolve flag values.")]
```

Reply:
Yes, after a quick look at the spec I found some potential mappings - I will go through the spec in detail and add them wherever it is applicable